aiosmtpd/docs/smtp.rst

```diff
@@ -104,31 +107,82 @@ override to provide additional responses.
 Handler hooks
 =============
 
-Handlers can implement a few hooks that get called during the ``SMTP`` dialog,
-or in exceptional cases.  More hooks may be added in the future, but for now
-the following handler hooks are defined:
+Handlers can implement a hooks that get called during the ``SMTP`` dialog, or
+in exceptional cases.  These *handler hooks* are all called asynchronously and
+they *must* return a status string, since the default statuses are *not*
+returned when a hook is defined.  Individual handlers may have additional
+responsibilities to replace default behavior, as described below.
+
+All handler hooks take at least three arguments, the ``SMTP`` server instance,
+:ref:`a session instance, and an envelope instance <sessions_and_envelopes>`.
+Some methods take additional arguments.
+
+The following hooks are currently defined:
+
+``handle_HELO(server, session, envelope, hostname)``
+    Called during ``HELO``, the ``hostname`` argument is the host name given
+    by the client in the ``HELO`` command.  If implemented, this hook must
+    also set the ``session.host_name`` attribute.
+
+``handle_EHLO(server, session, envelope, hostname)``
+    Called during ``EHLO``, the ``hostname`` argument is the host name given
+    by the client in the ``EHLO`` command.  If implemented, this hook must
+    also set the ``session.host_name`` attribute.  This hook may push
+    additional ``250-<command>`` responses to the client by yielding from
+    ``server.push(status)``.
```

Comment:
I suggest adding " before returning ``250 OK`` as the final response." at the end to clarify that `handle_EHLO` should still return the final response.